src/MMU.cpp

```diff
@@ -102,6 +114,19 @@ void MMU::writeDirectionButton(uint8_t pos, uint8_t value) {
 
 void MMU::writeButton(uint8_t* buttons, uint8_t pos, uint8_t value) {
 	// Create mask for the bit at pos, then or with the value at pos
+	uint8_t oldButtons = *buttons;
 	*buttons &= ~(1 << pos);
 	*buttons |= (value << pos);
+
+	// If a bit went from high to low, request a joypad interrupt
+	if (oldButtons > *buttons) {
+		setBit(0xFF0F, 4, 1);
+	}
+}
+
+void MMU::setBit(uint16_t addr, uint8_t pos, uint8_t value) {
```

Comment:
I think there's other places where I can use setBit, but sometimes its probably faster to only do one read and write if I need to set multiple bits at a time. This should only be used for setting one single bit.